pysat/_orbits.py

```diff
@@ -289,7 +289,15 @@ def __iter__(self):
         while True:
             try:
                 self.next()
-                yield self.inst
+
+                # Ensure that garbage collection doesn't delete self.inst
+                # by yielding a copy, without spending time on copying data
+                data = self.inst.data
+                self.inst.data = self.inst._null_data
+                local_inst = self.inst.copy()
```

Comment:
I am worried that this will make the orbit iteration even slower than it already is.  It may be necessary, but I think it would be a good idea to do a time test.

Reply:
Fair enough, here's a time test.
With the code modification:
```
In [42]: %timeit for inst in test.orbits: pass 
    ...:                                                                                                                                                                                                                    
pysat INFO: typical lt diff 0.004123711340206171
pysat INFO: Loaded Orbit: 0
pysat INFO: Loaded Orbit: 1
pysat INFO: Loaded Orbit: 2
pysat INFO: Loaded Orbit: 3
...
3.63 ms ± 201 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

Second run: 3.46 ms ± 239 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

```

The copy alone:
```
In [43]: test = pysat.Instrument('pysat', 'testing', orbit_info={'index': 'mlt'})                                                                                                                                           
pysat DEBUG: These Instrument attributes kept their default  values: ['directory_format', 'file_format', 'multi_file_day', 'orbit_info', 'pandas_format']
pysat DEBUG: These Instrument test attributes kept their default  values: ['_test_download', '_test_download_travis', '_password_req']
pysat INFO: pysat is searching for pysat testing files.
pysat INFO: test_list_files_kwarg = None
pysat INFO: Found 1096 local files.
pysat INFO: Test instruments provided through the pysat project. https://www.github.com/pysat/pysat

In [44]: %timeit test.copy()                                                                                                                                                                                                
502 µs ± 16.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Finally, the same orbit iteration sequence without the new code.
```
In [1]: import pysat                                                                                                                                                                                                        

In [2]: test = pysat.Instrument('pysat', 'testing', orbit_info={'index': 'mlt'})                                                                                                                                            

In [3]: import datetime as dt                                                                                                                                                                                               

In [4]: test.bounds = (dt.datetime(2008, 1, 1), dt.datetime(2008, 1, 31))                                                                                                                                                   

In [5]: pysat.logger.setLevel(1)                                                                                                                                                                                            

In [6]: %timeit for inst in test.orbits: pass                                                                                                                                                                               
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2008
pysat INFO: typical lt diff 0.004123711340206171
pysat INFO: Loaded Orbit: 0
pysat INFO: Loaded Orbit: 1
pysat INFO: Loaded Orbit: 2

3.68 ms ± 142 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

Second run: 3.36 ms ± 217 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

```

Reply:
I tested the load time itself,
```
In [7]: %timeit test.load(2009, 1)                                                                                                                                                                                          
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
pysat INFO: test_load_kwarg = None
pysat INFO: Returning pysat testing data for 01 January 2009
287 ms ± 5.16 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
So the orbits determination adds about 0.1 seconds to get a particular orbit.